### DISCUSSION TITLE
How to prevent automatically adding links where they are not needed?

### DISCUSSION THREAD

**Discussion by ilg-ul** [facebook/docusaurus#9649] [Q&A] [answered]
I noticed that sometimes words like `WWW` occurring in a markdown page, are generated as links, for example:

```
<a href="http://WWW" target="_blank" rel="noopener noreferrer">WWW</a>
```

This is obviously incorrect, and links validators complain.

Is there a way to prevent this?

**Comment by slorber** [comment]
@ilg-ul please create a runnable docusaurus.new repro showing the problem, because your repro instructions are not clear enough for me to reproduce it.

<img width="524" alt="CleanShot 2023-12-21 at 13 04 24@2x" src="https://github.com/facebook/docusaurus/assets/749374/a39a1338-2f21-490b-aac1-bc90b0d67ac8">

---

Some feedback: you opened many issues/discussions recently but we never had a repro in it as far as I remember. 
I only have limited time to work on Docusaurus, and doing so always leads me spend time trying to figure out your problem. 

When I support you by creating a repro attempt myself, this is time that I finally don't spend working on more important Docusaurus features, which means it's not time well-invested from my perspective.

For these reasons, and to be fair with all the Docusaurus community, I won't be able to support you in upcoming support requests unless a repro is provided.
I hope you understand my position, but I have to protect my time by asking you to invest yours instead, otherwise this project will not progress much.

**Comment by ilg-ul** [comment]
As instructed, I created a simple repo, and I added a few lines at the end of the welcome blog post:

```
---

- www. This generates a link.
- www . This does not.
- wwww. This does not.
- www.some.thing. This generates a link.
- www.some.thing . This generates a link.

Apparently the `www.` combination triggers this behaviour.
```

- https://github.com/ilg-ul/docusaurus-9649/blob/0e2b4a8f2dec8e40408586186e205862613edb96/website/blog/2021-08-26-welcome/index.md?plain=1#L27-L35

In my configuration, this page is rendered as:

![Screenshot 2023-12-21 at 23 42 33](https://github.com/facebook/docusaurus/assets/362732/205b5e12-2818-4340-bfce-f4e0e203e070)

Is there any way to prevent this behaviour?

Sorry for not providing the test case from the very beginning.

**Reply by Josh-Cena** [reply]
The best way is perhaps escaping: `www\.` Because MDX replicates behavior of GFM you can test on GitHub itself.

> www.
> www\.

Perhaps `<span>www.</span>` works too? https://stackoverflow.com/questions/25706012/how-do-i-prevent-auto-generated-links-in-the-github-wiki looks relevant, although it's about GFM, which is an HTML superset not a JSX superset, so tricks like void tags don't work well here.

**Comment by ilg-ul** [comment]
Based on your suggestions, I did some more tests:

```
- www. This generates a link.
- www\. This generates a link.
- www . This does not.
- wwww. This does not.
- <span>www</span>. This does not.
- www<span></span>. This does not.
- www<!-- -->. This generates a link.
- www&period; This generates a link.
- www.some.thing. This generates a link.
- www.some.thing . This generates a link.
- www\.some.thing. This generates a link.
- www<!-- -->.some.thing. This generates a link.
- www&period;some.thing. This generates a link.
```

Do you foresee any issues with something like `www<span></span>.`?

**Reply by Josh-Cena** [reply]
No; that looks sufficiently safe to me.

**Comment by ilg-ul** [comment, marked as answer]
Thank you, then I'll use the `www<span></span>.` syntax to prevent auto-linking.

**Reply by slorber** [reply]
Great that you found a solution

If you want to know if it's a Docusaurus bug or a MDX + GFM behavior that we choose to not diverge from, you can use the MDX playground, or eventually try your markdown on GitHub too.

https://mdxjs.com/playground/

![CleanShot 2023-12-22 at 12 24 11@2x](https://github.com/facebook/docusaurus/assets/749374/09d5286f-203a-416a-84b2-eb13b726cddd)

If MDX or GitHub also implement the behavior you don't like, then it should be reported to MDX or GitHub directly

**Reply by ilg-ul** [reply]
Thank you for clarifying it!

The actual question was how to prevent it, and I think it got an acceptable answer.
